src/ZoneServer/Buffs/Buff.cs

```diff
@@ -213,11 +213,15 @@ public Buff(BuffId buffId, float numArg1, float numArg2, TimeSpan duration, Time
 
 		/// <summary>
 		/// Increase overbuff counter, capped to the buff's max overbuff
-		/// value.
+		/// value.  Also reset the buff's update time, if it has one
 		/// </summary>
 		public void IncreaseOverbuff()
 		{
 			this.OverbuffCounter++;
+			this.Handler?.OnOverbuff(this);
+
+			if (this.HasUpdateTime)
```

Comment:
I definitely think it makes more sense for the handler to be here, if you mean specifically the code related to the update time it could be moved, but it requires the protection level to be changed since NextUpdateTime is private.